test/drlg_l1_test.cpp

```diff
@@ -19,3 +26,134 @@ TEST(Drlg_l1, DRLG_Init_Globals)
 	DRLG_Init_Globals();
 	EXPECT_EQ(dLight[0][0], 0);
 }
+
+void TestCreateL5Dungeon(bool hellfire, int level, int seed, lvl_entry entry)
+{
+	pMegaTiles = std::make_unique<MegaTile[]>(1648);
+
+	MyPlayer->pOriginalCathedral = !hellfire;
+
+	currlevel = level;
+	leveltype = DTYPE_CATHEDRAL;
+
+	CreateL5Dungeon(seed, entry);
+
+	std::string path = paths::BasePath();
+
+	paths::SetPrefPath(path);
+	std::string dunPath;
+	if (hellfire)
+		dunPath = fmt::format("../test/fixtures/hellfire/{}-{}.dun", level, seed);

```

Comment:
Probably need to add the ability to choose files based on quest state but I wouldn't worry about that for now :D